usage/test-framework/src/main/java/org/apache/brooklyn/test/framework/BaseTest.java

```diff
@@ -0,0 +1,34 @@
+package org.apache.brooklyn.test.framework;
+
+import org.apache.brooklyn.api.entity.Entity;
+import org.apache.brooklyn.config.ConfigKey;
+import org.apache.brooklyn.core.config.ConfigKeys;
+import org.apache.brooklyn.core.entity.trait.Startable;
+import org.apache.brooklyn.util.core.flags.SetFromFlag;
+
+/**
+ *  A base interface for all tests.
+ */
+public interface BaseTest extends Entity, Startable {
+
+    /**
+     * The target entity to test (optional, use either this or targetId).
+     */
+    @SetFromFlag(nullable = false)
```

Comment:
i now see it's very useful to pass the `Entity` when writing in java; we should definitely continue to support this.